tests/robustness/main_test.go

```diff
@@ -158,3 +167,30 @@ func forcestopCluster(clus *e2e.EtcdProcessCluster) error {
 	}
 	return clus.ConcurrentStop()
 }
+
+func checkForCompaction(clus *e2e.EtcdProcessCluster) (bool, error) {
+	req := e2e.CURLReq{
+		Endpoint: "/metrics",
+		Expected: expect.ExpectedResponse{
+			// Filter out the etcd_debugging_mvcc_db_compaction_keys_total
+			// metric if it has a value greater than 0 from the response returned
+			// by the /metrics endpoint.
+			Value:         `etcd_debugging_mvcc_db_compaction_keys_total\s+[1-9][0-9]*`,
+			IsRegularExpr: true,
+		},
+	}
+
+	err := e2e.CURLGet(clus, req)
+	if err == nil {
+		return true, nil
+	}
+
+	// If no match was found, compaction did not take place.
+	noCompaction := strings.Contains(err.Error(), "match not found.") ||
+		strings.Contains(err.Error(), "context deadline exceeded")
```

Comment:
This feels odd to do - maybe its worth using `e2e.SpawnWithExpectsContext` and checking the len of the returned lines instead?

Reply:
Ah that makes sense

However, one issue I see with this is if the value of the metric is 0 vs is curl didn't succeed, I can't really differentiate there. If I had the output of `SpawnWithExpectsContext`, I could maybe error out when it makes sense.

Reply:
That's a good point, there is a chance that we metric is changed and as result we run the validator. However, the resulting validation will be more strict as not detecting compaction will cause prevKV to be required. This leaves possibility of false positives, however I think this is acceptable for short term solution.